core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -13,6 +13,31 @@
 import org.slf4j.LoggerFactory;
 import sun.misc.Unsafe;
 
+/**
+ * A monoid to generate histogram buckets along with the common aggregation functions for a given
+ * metric.
+ *
+ * <p>Performs the following aggregation functions on the measurements:</p>
+ * <ul>
+ * <li>count of measurements</li>
+ * <li>sum of the measurement values</li>
+ * <li>minimum measured value</li>
+ * <li>maximum measured value</li>
+ * <li>last measured value</li>
+ * </ul>
+ *
+ * The histogram buckets are derived from {@link DistributionBucket}.
+ * <p>A sample histogram would look like:</p>
+ * <ul>
+ * <li>[0_10)</li>
```

Comment:
While the buckets in this doc look pretty, we are likely going to have an ugly outcome with bucket naming for timer metrics because the unit of measurement is in nanoseconds while most humans think/expect in milliseconds.
Ideally, when the reporters are built, we should allow users to specify the output bucket spec unit so the devs can state they wish to see bucket names in milliseconds even though internally the library tracks them in nanoseconds.

Additionally, we should have an option to prettify the bucket names to avoid things like `52.4498324_58.230823832`, rather `52.5_58.2`, etc